examples/experimental/dagster-airlift/dagster_airlift/core/sensor/event_translation.py

```diff
@@ -24,6 +30,71 @@
 ]
 
 
+def default_event_transformer(
+    context: SensorEvaluationContext,
+    airflow_data: AirflowDefinitionsData,
+    materializations: Sequence[AssetMaterialization],
+) -> Iterable[AssetEvent]:
+    """The default event transformer function, which attaches a partition key to materializations which are from time-window partitioned assets."""
+    for mat in materializations:
+        asset_spec = airflow_data.get_asset_spec_for_key(mat.asset_key)
+        if not asset_spec.is_time_window_partitioned:
+            yield mat
+            continue
+        airflow_logical_date_timestamp: float = cast(
+            TimestampMetadataValue, mat.metadata[AIRFLOW_TASK_INSTANCE_LOGICAL_DATE_METADATA_KEY]
+        ).value
+        airflow_run_id = cast(str, mat.metadata[AIRFLOW_RUN_ID_METADATA_KEY])
+        partition = get_partition_key_from_existing_materialization(
+            partitions_def=cast(TimeWindowPartitionsDefinition, asset_spec.partitions_def),
+            airflow_logical_date_timestamp=airflow_logical_date_timestamp,
+            airflow_run_id=airflow_run_id,
+            asset_key=mat.asset_key,
+        )
+        partitioned_mat = mat._replace(partition=partition)
+        yield partitioned_mat
```

Comment:
this is the new bit. We need to transform existing materializations instead of applying information to new ones.